setup.py

```diff
@@ -1,11 +1,20 @@
 from distutils.core import setup
-from distutils.extension import Extension
 from glob import glob
 from os import environ as env, path
 import os
 import sys
 import numpy as np
 import petsc4py
+from firedrake_configuration import get_config
```

Comment:
This introduces a circular dependency in setup.py.  This file *installs* the `firedrake_configuration` module, yet relies on it already existing.  Oops!